augurpy/read_load.py

```diff
@@ -16,21 +22,53 @@ def load(input: Union[AnnData, DataFrame], meta: DataFrame, label_col: str, cell
         cell_type_col: column of the meta DataFrame or the Anndata or matrix containing the cell type labels for each
             cell in the cell-by-gene expression matrix
 
-
     Returns:
-        Pandas DataFrame containing gene expression values (cells in rows, genes in columns), cell type and condition
+        Anndata object containing gene expression values (cells in rows, genes in columns) and cell type, label as obs
     """
-    pass
+    if isinstance(input, AnnData):
+        input.obs = input.obs.rename(columns={cell_type_col: "cell_type", label_col: "label"})
+        out = input
+
+    elif isinstance(input, DataFrame):
+        try:
+            cell_type = input[cell_type_col]
+            label = input[label_col]
+        except KeyError:
+            print("No column names matching cell_type_col and label_col. Looking in meta data.")
+            try:
+                cell_type = meta[cell_type_col]
+                label = meta[label_col]
+            except (KeyError, TypeError):
+                raise Exception("Missing label and / or cell type column.") from None
+
+            else:
+                print("Adding cell type and label from metadata.")
+                out = AnnData(X=input, obs={"cell_type": cell_type, "label": label})
+
+        else:
+            out = AnnData(X=input, obs={"cell_type": cell_type, "label": label})
+
+    else:
+        raise Exception("Not valid input.")
 
+    out = feature_selection(out)
 
-def feature_selection(input: DataFrame, var_quantile: float) -> DataFrame:
+    return out
+
+
+def feature_selection(input: AnnData) -> AnnData:
```

Comment:
Do you expect there to be more feature selection methods later?
If so, we could either parameterize this method with the specific feature selection?

Reply:
Originally I thought it would be more complicated which is why I put it into a separate function. Im not sure if there will be more feature selection methods - there are a couple different modes in scanpy, we are now only using the defaults. Maybe it would make sense to pass that on as a parameter. Not sure I understand what you are suggesting.

Reply:
Basically, it could look like:

```
def feature_selection(input: AnnData, method: str = "highly_variable") -> AnnData

where method could be highly_variable or many others.

But we can also refactor this later if we come up with novel feature selection methods.

Reply:
Right! Lets do that.